typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,440 @@
+                if (exists != null) {
+                    return new Error("The Kurtosis API asked us to copy static file " + staticFileId + " to path " + destFilepathRelativeToEnclaveVolRoot + 
+                    " in the enclave volume which means that an empty file should exist there, " + "but no file exists at that path - this is a bug in Kurtosis!");
+                }
+            })
+            //TODO TODO TODO
+            // srcFp, err := os.Open(srcAbsFilepath)
+            // if err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred opening static file '%v' source file '%v' for reading", staticFileId, srcAbsFilepath)
+            // }
+            // defer srcFp.Close()
+
+            // destFp, err := os.Create(destAbsFilepath)
+            // if err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred opening static file '%v' destination file '%v' for writing", staticFileId, destAbsFilepath)
+            // }
+            // defer destFp.Close()
+
+            // if _, err := io.Copy(destFp, srcFp); err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred copying all the bytes from static file '%v' source filepath '%v' to destination filepath '%v'", staticFileId, srcAbsFilepath, destAbsFilepath)
+            // }
+
+        }
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public registerFilesArtifacts(filesArtifactUrls: Map<FilesArtifactID, string>): Error {
+        const filesArtifactIdStrsToUrls: Map<string, string> = new Map();
+        for (let artifactId in filesArtifactUrls) {
+            const url: string = filesArtifactUrls[artifactId];
+            filesArtifactIdStrsToUrls[String(artifactId)] = url;
+        }
+        const args: RegisterFilesArtifactsArgs = newRegisterFilesArtifactsArgs(filesArtifactIdStrsToUrls);
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // if _, err := networkCtx.client.RegisterFilesArtifacts(context.Background(), args); err != nil {
+        //     return stacktrace.Propagate(err, "An error occurred registering files artifacts: %+v", filesArtifactUrls)
+        // }
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public addService(
+        serviceId: ServiceID,
+        containerCreationConfig: ContainerCreationConfig, //TODO
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error]
+    ): [ServiceContext, Map<string, PortBinding>, Error] { //TODO
+
+        const [serviceContext, hostPortBindings, err] = this.AddServiceToPartition(
+            serviceId,
+            defaultPartitionId,
+            containerCreationConfig,
+            generateRunConfigFunc,
+            )
+        if (err != null) {
```

Comment:
`!==` rather than `!=` - make sure you check for this everywhere in all your PRs, since the consequences of getting it wrong can be subtle and reallyyyyyy hard to debug.

(related rant: this is an example of terrible language design, where what the programming _thinks_ would make sense, `!=`, will actually smack them and the programmer instead needs to a) know the problem and b) remember it every. single. time.)